### DISCUSSION TITLE
URL structure limits possibility of viewing all other publisher's packages

### DISCUSSION THREAD

**Discussion by yairEO** [npm/feedback#283] [Website]
### For example, a (scoped) package page URL:

    https://www.npmjs.com/package/@yaireo/tagify
    
A user would ***expect*** to be able to view all the packages of `@yaireo` by simply modifying the URL as so:

    https://www.npmjs.com/package/@yaireo/
                         ‾‾‾‾‾‾‾‾
        
But this does not work, resulting in `404` page. The word `package` in the URL is IMHO incorrect, and should be named in plural, `packages`, since a person or org can have multiple, and the last part of the path is obviously the specific package (name)

---

I would also expect that if a package is a part of an scoped org, the org name would be a list to view all the org's public packages. As shown below, clicking on the part marked in  *red*  should redirect to the package list page or that org

![image](https://user-images.githubusercontent.com/845031/111062633-b32c4480-84b2-11eb-84c6-ff6148a05150.png)

**Comment by MylesBorins** [comment]
hey @yairEO 

It seems like the biggest request here is to make it more intuitive to find / list the packages that are within a scope and you have a couple specific suggestions on how we could do this

1. that /package/:scope should be able to redirect to the scope landing page
2. the org name on a package landing page should be clickable and forward to the scope landing page

Separate from this you pointed out that /package/:scope doesn't seem accurate and it should instead be /packages/:scope... or at least it could forward to that. Of all the points you've raised I think this might be the hardest to address as we need to think about canonical URLs and the cost of redirection.

I'm going to open an issue with the team and see what might make sense as a way to improve the experience

**Reply by yairEO** [reply]
right, but what is your own point of view regarding to what I said? do you agree?

**Reply by MylesBorins** [reply]
@yairEO I agree that the current experience for finding the list of all packages in a scope is harder than it should be.

It is not clear to me right now which solution is the best to move forward with because I'm not sure how much engineering effort each would require.
